src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/SlicesController.cs

```diff
@@ -0,0 +1,148 @@
+using ProjectOrigin.WalletSystem.Server.Database;
+using ProjectOrigin.WalletSystem.Server.Extensions;
+using ProjectOrigin.WalletSystem.Server.Models;
+
+namespace ProjectOrigin.WalletSystem.Server.Services.REST.v1;
+
+[AllowAnonymous]
+[ApiController]
+public class SlicesController : ControllerBase
+{
+    /// <summary>
+    /// Receive a certificate-slice from another wallet.
+    /// </summary>
+    /// <remarks>
+    /// This request is used to receive a certificate-slice from another wallet, which is then stored in the local wallet.
+    /// The endpoint is verified to exists within the wallet system, otherwise a 404 will be returned.
+    /// The endpoint will return 202 Accepted was initial validation has succeeded.
+    /// The certificate-slice will further verified with data from the registry in a seperate thread.
+    /// </remarks>
+    /// <param name = "unitOfWork" ></param>
+    /// <param name = "hdAlgorithm" ></param>
+    /// <param name = "bus" ></param>
+    /// <param name = "request" >Contains the data </param>
+    /// <response code="202">The slice was accepted.</response>
+    /// <response code="400">Public key could not be decoded.</response>
+    /// <response code="404">Receiver endpoint not found.</response>
+    [HttpPost]
+    [Route("v1/slices")]
+    [Produces("application/json")]
+    [ProducesResponseType(StatusCodes.Status202Accepted)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<ReceiveResponse>> ReceiveSlice(
+        [FromServices] IUnitOfWork unitOfWork,
+        [FromServices] IHDAlgorithm hdAlgorithm,
+        [FromServices] IBus bus,
+        [FromBody] ReceiveRequest request
+    )
+    {
+        if (!hdAlgorithm.TryImportHDPublicKey(request.PublicKey, out var publicKey))
+            return BadRequest("Invalid public key.");
+
+        var endpoint = await unitOfWork.WalletRepository.GetWalletEndpoint(publicKey);
+        if (endpoint == null)
+            return NotFound("Endpoint not found for public key.");
+
+        var newSliceCommand = new VerifySliceCommand
+        {
+            Id = Guid.NewGuid(),
+            WalletId = endpoint.WalletId,
+            WalletEndpointId = endpoint.Id,
+            WalletEndpointPosition = (int)request.Position,
+            Registry = request.CertificateId.Registry,
+            CertificateId = request.CertificateId.StreamId,
+            Quantity = request.Quantity,
+            RandomR = request.RandomR,
+            HashedAttributes = request.HashedAttributes.Select(x => new WalletAttribute
+            {
+                CertificateId = request.CertificateId.StreamId,
+                RegistryName = request.CertificateId.Registry,
+                Key = x.Key,
+                Value = x.Value,
+                Salt = x.Salt
+            }).ToList()
+        };
+
+        await bus.Publish(newSliceCommand);
```

Comment:
Should this be awaited?

Reply:
Yes, it awaits it being published to the broker, otherwise you don't know if it will succeed or not. The invocation of the command is started separately by mass transit.